src/app/market/buy/checkout-process/checkout-process.component.ts

```diff
@@ -214,8 +214,11 @@ export class CheckoutProcessComponent implements OnInit, OnDestroy {
     })
   }
 
-  select(address: Address) {
+  select(address: Address, isNewProfile?: boolean) {
     this.log.d('Selecting address with id: ' + address.id);
+    if (isNewProfile) {
+      this.profile.shippingAddresses = [address];
```

Comment:
When navigating from step 3 to 2 int he checkout process, the shipping profile is now correctly selected, if one was previously selected. However, it is the only option in the shipping profile dropdown, which I suspect is a consequence of this specific line of code.